lib/maths/CBoostedTreeImpl.cc

```diff
@@ -1628,6 +1955,22 @@ void CBoostedTreeImpl::skipProgressMonitoringFinalTrain() {
     m_Instrumentation->startNewProgressMonitoredTask(CBoostedTreeFactory::FINAL_TRAINING);
 }
 
+void CBoostedTreeImpl::startProgressMonitoringTrainIncremental() {
+
+    // This costs "number folds" * "maximum number retrained trees" units
+    // per round.
+
+    m_Instrumentation->startNewProgressMonitoredTask(CBoostedTreeFactory::INCREMENTAL_TRAIN);
+
+    std::size_t totalNumberSteps{m_NumberRounds * this->numberTreesToRetrain() * m_NumberFolds};
+    LOG_TRACE(<< "main loop total number steps = " << totalNumberSteps);
+    m_TrainingProgress = core::CLoopProgress{
+        totalNumberSteps, m_Instrumentation->progressCallback(), 1.0, 1024};
+
+    // Make sure progress starts where it left off.
+    m_TrainingProgress.increment(m_CurrentRound * this->numberTreesToRetrain() * m_NumberFolds);
+}
+
 namespace {
 const std::string VERSION_7_11_TAG{"7.11"};
```

Comment:
Should we add 8.0 version tag here? Or a todo to adjust to whatever version will be current when we merge to master? Otherwise we will probably forget about it by the time.

Reply:
Yes. Let's add a TODO. It is probably possible to upgrade state, but we can revisit when we have finalised everything.

Reply:
See [0472b2a](https://github.com/elastic/ml-cpp/pull/1837/commits/0472b2a7a4fa67310d0058f8cde208b0642bda63).